yt_dlp/extractor/tvopengr.py

```diff
@@ -0,0 +1,150 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+import urllib.parse
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    get_elements_text_and_html_by_attribute,
+    unescapeHTML,
+)
+
+
+class TVOpenGrBaseIE(InfoExtractor):
+    def _return_canonical_url(self, url, video_id):
+        webpage = self._download_webpage(url, video_id)
+        canonical_url = self._og_search_url(webpage)
+        title = self._og_search_title(webpage)
+        self.to_screen(f'{video_id}: Redirect to canonical URL')
```

Comment:
is this necessary?

Reply:
Not really, it just seemed good to explain what was happening. I will remove it.